docs/extend/update-b14.md

```diff
@@ -0,0 +1,98 @@
+Due to the nature and size of these upgrades, we have to pass on some of the breaking changes to you, our extension developers.
+
+On the bright side, this overdue upgrade brings us closer to the conventions of best practices of [Mithril.js](https://mithril.js.org/), the mini-framework used for Flarum's UI.
+Mithril's 2.0 release sports a more consistent component interface, which should be a solid foundation for years to come.
+Where possible, we replicated old APIs, to ease the upgrade and give you time to do the full transition.
+Quite a few breaking changes remain, though - read more below.
+
+::: tip
+If you need help with the upgrade, our friendly community will gladly help you out either [on the forum](https://discuss.flarum.org/t/extensibility) or [in chat](https://flarum.org/chat/).
+:::
+
+To ease the process, we've clearly separated the changes to the frontend (JS) from those in the backend (PHP) below.
+If your extension does not change the UI, consider yourself lucky. :-)
+
+## Frontend (JavaScript)
+
+### Changes in Core
+
+*TODO: States, and other BC breaks*
+
+### Mithril 2.0: Concepts
+
+*TODO: Explain*
+
+- props -> attrs
+- vnodes
+- Component instances should not be stored
+
+### How to upgrade a component
+
+#### Required changes
+
+*TODO*
+
+- `view()` -> `view(vnode)`
+- Lifecycle
+  - `init()` -> `oninit(vnode)`
+  - `config()` -> Lifecycle hooks `oncreate(vnode)` / `onupdate(vnode)`
+  - `context.onunload()` -> `onremove()`
+  - `SubtreeRetainer` -> `onbeforeupdate()`
+- if present, `attrs()` method needs to be renamed -> convention `elementAttrs()`
+- building component with `MyComponent.component()` -> `children` is now second parameter instead of a named prop/attr (first argument) -> JSX preferred
+- Routing
+  - `m.route()` -> `m.route.get()`
+  - `m.route(name)` -> `m.route.set(name)`
+  - register routes with page class, not instance
+    - special case when passing props
+  - `<a href={url} config={m.route}>` -> `<a route={url}>`
+- AJAX
+  - `m.request({...})` -> `data:` key split up into `body:` and `params:`
+  - `m.deferred` -> native `Promise`
+- Redrawing
+  - `m.redraw(true)` -> `m.redraw.sync()`
+  - `m.redraw.strategy('none')` -> `e.redraw = false` in event handler
+  - `m.lazyRedraw()` -> `m.redraw()`
+
+#### Optional changes
+
+For the following changes, we currently provide a backwards-compatibility layer.
+This will be removed in time for the stable release.
+The idea is to let you release a new version that's compatible with Beta 14 to your users as quickly as possible.
+When you have taken care of the changes above, you should be good to go.
+For the following changes, we have bought you time until the stable release.
+Considering you have to do the changes anyway, why not do them now?
+
+*TODO*
+
+- `this.props` -> `this.attrs`
+- static `initProps()` -> static `initAttrs()`
+- `m.prop` -> `m.stream`
+- `m.withAttr` -> `withAttr` with import
```

Comment:
This is my proposal for a minimal BC layer. These... 

a) are trivial to add and maintain,
b) will be removed for the stable release, and
c) are repetitive and frequent, so avoiding them now should help the extension ecosystem.

Reply:
Not sure what you mean with this comment? Do you want to remove all of these?